airbyte/_util/telemetry.py

```diff
@@ -137,9 +158,11 @@ def get_env_flags() -> dict[str, Any]:
 
 
 def send_telemetry(
-    source: Source,
+    source: Source | None,
```

Comment:
Source param can't be mandatory anymore since we don't have enough information to create a Source in most cases when install failures happens. Instead, a source name is passed in those cases.

Reply:
Wdyt or re-typing `source` as `Source | str`? (For now, at least, and later `Source | str | None` when/if we have events that don't have any source at all?

In this case, we'd replace the below null checks with `isinstance()` checks - taking one code path if `Source` is `str` and another if it's an object.

Reply:
oh yea, makes a lot of sense! updated.